source/common/access_log/access_log_formatter.cc

```diff
@@ -54,24 +56,72 @@ AccessLogFormatUtils::protocolToString(const absl::optional<Http::Protocol>& pro
 }
 
 FormatterImpl::FormatterImpl(const std::string& format) {
-  formatters_ = AccessLogFormatParser::parse(format);
+  providers_ = AccessLogFormatParser::parse(format);
 }
 
 std::string FormatterImpl::format(const Http::HeaderMap& request_headers,
                                   const Http::HeaderMap& response_headers,
-                                  const Http::HeaderMap& response_trailers,
+                                  const Http::HeaderMap& response_trailers, 
                                   const RequestInfo::RequestInfo& request_info) const {
   std::string log_line;
   log_line.reserve(256);
 
-  for (const FormatterPtr& formatter : formatters_) {
+  for (const FormatterProviderPtr& provider : providers_) {
     log_line +=
-        formatter->format(request_headers, response_headers, response_trailers, request_info);
+        provider->format(request_headers, response_headers, response_trailers, request_info);
   }
 
   return log_line;
 }
 
+JsonFormatterImpl::JsonFormatterImpl(const std::map<const std::string, const std::string>& format_mapping) {
+  for (const auto& format_pair : format_mapping) {
```

Comment:
other loops in this file don't seem to use `auto`, maybe the agreement is to spell out the type?

Reply:
Kept this as auto and changed others to also be auto